### ISSUE TITLE
Add min pooling support via negate-max-negate

### ISSUE THREAD

**Issue by tillahoffmann** [tillahoffmann/jax-mps#61]
## Summary

Min pooling is currently not supported in `reduce_window`. MPS pooling only provides `maxPooling4D` and `avgPooling4D`, but min pooling can be implemented as:

```
min(x) = -max(-x)
```

## Implementation

In `HandlePoolingReduceWindow` (reduction_ops.mm):

1. Detect `stablehlo.minimum` reduction type
2. Negate the input tensor
3. Apply max pooling
4. Negate the result

This is a common mathematical identity that allows reusing the existing max pooling infrastructure.

## Test

```python
lax.reduce_window(x, jnp.inf, lax.min, (1, 2, 2, 1), (1, 2, 2, 1), "valid")
```

**Comment by jkitchin** [comment]
This looks pretty easy to address, but I found that "The 14 skips are grad tests, correctly skipped since select_and_scatter isn't supported yet." Max and min pool forward passes work. Gradients for both require select_and_scatter, which is a separate unimplemented op (stablehlo.select_and_scatter).
  MPS provides maxPooling4DGradientWithGradientTensor:sourceTensor:descriptor:name: which maps directly to it. Min pool grad would use the same  negate-max-negate pattern on top of that.

Would you prefer a total fix here, or a new issue for select_and_scatter?
